### ISSUE TITLE
Bugs when running Linf attack on pt_inception

### ISSUE THREAD

**Issue by Daizy97** [max-andr/square-attack#11]
Thanks for your great job!  I have some trouble running your code.
I want to run your code on docker server. But since you use both Tensorflow and Pytorch, I could not find a proper docker image and encountered a lot of version errors. Finally I gave up and try to remove the Tensorflow part in your code (remove some irrelevant codes related to TFmodels in models.py) and just want to test on pt_inception model. But still I got some bug:
![image](https://user-images.githubusercontent.com/30400618/109614480-a7b15300-7b6d-11eb-9743-0ab9b366d584.png)
I appreciate if you can help me!

**Comment by max-andr** [comment]
Hi,

Thanks for the interest in our paper!
It's a bit hard to tell what can be a problem here without having your version of the code. But I think you should just print `logits_list` and see why it's not possible to vertically concatenate it. Most likely, the `model.predict()` function in your version of the code returns something unexpected while it should return a matrix of logits of size `number_of_samples x number_of_classes`.

I hope that helps. Feel free to reopen the issue if there are more questions on this!

**Comment by Daizy97** [comment]
> Hi,
> 
> Thanks for the interest in our paper!
> It's a bit hard to tell what can be a problem here without having your version of the code. But I think you should just print `logits_list` and see why it's not possible to vertically concatenate it. Most likely, the `model.predict()` function in your version of the code returns something unexpected while it should return a matrix of logits of size `number_of_samples x number_of_classes`.
> 
> I hope that helps. Feel free to reopen the issue if there are more questions on this!

Thanks for your valuable advice. It seems the problem comes from loading ImageNet. If possible, could you show me how your files arranged in this path `IMAGENET_PATH = "/scratch/maksym/imagenet/val_orig"` (Line 29, data.py)? Are val images and labels in the same directory?
My file is arranged like this:
![image](https://user-images.githubusercontent.com/30400618/109905185-5cb44e80-7cd9-11eb-9359-7f6e1d6ed815.png)
But I found after loading the dataset, the `y_test` is all zeros.
Thanks very much for your help!

**Comment by max-andr** [comment]
Hi,

So in the directory `IMAGENET_PATH = "/scratch/maksym/imagenet/val_orig"`, you should have folders named like `n01440764`, ` n01739381`, `n01978287`, etc. In each folder there are images of a particular class. 

For further details, you can check the documentation of `ImageFolder`:
https://pytorch.org/vision/0.8/datasets.html#imagefolder

I hope that helps!

**Comment by Daizy97** [comment]
> Hi,
> 
> So in the directory `IMAGENET_PATH = "/scratch/maksym/imagenet/val_orig"`, you should have folders named like `n01440764`, ` n01739381`, `n01978287`, etc. In each folder there are images of a particular class.
> 
> For further details, you can check the documentation of `ImageFolder`:
> https://pytorch.org/vision/0.8/datasets.html#imagefolder
> 
> I hope that helps!

Thanks very much for your help!
